### PR TITLE
Fix/contractkit test flakyness

### LINKED ISSUE #1996: Make ContractKit tests not flaky
### Expected Behavior

No flaky tests

### Current Behavior

Flaky contractkit tests

### PR DESCRIPTION
**Pull request by H34D**
### Description

- Sequencing test runs alphabetic
- Add more timeout to long running tests
- Properly log ganache startup
- Give ganache some time to start before running first tests

### Tested

ran 5 times:
- yarn test:reset
- yarn test

all tests were green

Closes: #1996 #1773

### CODE REVIEW

File: packages/contractkit/src/wrappers/Accounts.test.ts
Comment (jmrossy) [inline packages/contractkit/src/wrappers/Accounts.test.ts]: 100 seconds? seems high

**Review by jmrossy — COMMENTED** [review]
Mostly looks good! one comment below. Recommend waiting for @mcortesi approval since he knows this package best :)

File: packages/contractkit/src/wrappers/Accounts.test.ts
Comment (H34D) [reply packages/contractkit/src/wrappers/Accounts.test.ts]: whoops, wanted 10 seconds

File: packages/contractkit/src/test-utils/AlphabeticSequencer.ts
Comment (mcortesi) [inline packages/contractkit/src/test-utils/AlphabeticSequencer.ts]: why the `ts-ignore`?

File: packages/contractkit/src/test-utils/AlphabeticSequencer.ts
Comment (mcortesi) [inline packages/contractkit/src/test-utils/AlphabeticSequencer.ts]: Why do we need a sequencer? I would put some documentation about why & how it works... or what it solves

File: packages/contractkit/src/test-utils/ganache.setup.ts
Comment (mcortesi) [inline packages/contractkit/src/test-utils/ganache.setup.ts]: 10 seconds seems a bit excessive, but not sure what was the problem here, again maybe document why we do this...

**Review by mcortesi — COMMENTED** [review]
There's a `dev-utils` package that @nambrot created, i though contractkit was already using that, since it contains ganache setup and other test utilities.

Maybe we should migrate contractkit to use it, and do the changes there. For now, `celocli` use the same setup, and the attestantion service i believe.

File: packages/contractkit/src/test-utils/ganache.setup.ts
Comment (H34D) [reply packages/contractkit/src/test-utils/ganache.setup.ts]: Ganache needs some time to start up. I can put this to 5 seconds maybe but then it might fail again from time to time. With 10 seconds it did not fail a single time in my tests. 

Where exactly shall i document this? Inline or in a separate document?

File: packages/contractkit/src/test-utils/AlphabeticSequencer.ts
Comment (H34D) [reply packages/contractkit/src/test-utils/AlphabeticSequencer.ts]: By default jest uses its internal sequencer like this:

https://github.com/facebook/jest/blob/master/packages/jest-test-sequencer/src/index.ts#L83-L110

this sequencer has some logic built in to re-run and re-prioritise test runs. This leads to a different order on every single test run. With an alphabetic order we get a more deterministic testrun than with the built in sequencer. It is more reproducible this way. 

The general problem is that we are using jest for blockchain tests, this is not the best technology fit because it was made for heavy parallelization. This is already being mitigated with the `--runInBand` flag sequencing the tests over all.

Where exactly shall i document this? Inline or in a separate document?

File: packages/contractkit/src/test-utils/AlphabeticSequencer.ts
Comment (H34D) [reply packages/contractkit/src/test-utils/AlphabeticSequencer.ts]: > There's a `dev-utils` package that @nambrot created, i though contractkit was already using that, since it contains ganache setup and other test utilities.
> 
> Maybe we should migrate contractkit to use it, and do the changes there. For now, `celocli` use the same setup, and the attestantion service i believe.

Amazing idea, i will open an issue for it.

Comment (H34D) [reply packages/contractkit/src/test-utils/AlphabeticSequencer.ts]: > why the `ts-ignore`?

migrated sequencer back to js, removed the ts-ignore

**Comment by H34D** [comment]
> There's a `dev-utils` package that @nambrot created, i though contractkit was already using that, since it contains ganache setup and other test utilities.
> 
> Maybe we should migrate contractkit to use it, and do the changes there. For now, `celocli` use the same setup, and the attestantion service i believe.

~Amazing idea, i will open an issue for it.~

After looking at the code again, it is using `dev-utils` already this is just the adapter code. All fine.

`import baseSetup from '@celo/dev-utils/lib/ganache-setup'`

File: packages/contractkit/src/test-utils/ganache.setup.ts
Comment (asaj) [reply packages/contractkit/src/test-utils/ganache.setup.ts]: PTAL @nategraf 's recent PR that added some tooling to spin until ganache comes up. Seems like it should be re-used here

https://github.com/celo-org/celo-monorepo/pull/2104/files#diff-7c886cc81e742d2a57cd84d6a6e42e58R137

Comment (H34D) [reply packages/contractkit/src/test-utils/ganache.setup.ts]: - carved out waitForPortOpen from celotool and moved it to dev-utils
- migrated ganache setup from sleep to waitForPortOpen from dev-utils
in https://github.com/celo-org/celo-monorepo/pull/2150/commits/16e049ff180630cbaac7744869c2842f254c4fe5

File: packages/contractkit/src/test-utils/AlphabeticSequencer.ts
Comment (mcortesi) [reply packages/contractkit/src/test-utils/AlphabeticSequencer.ts]: Thanks for the explanation...
I would document as a comment what you just said...  for future maintainers to know why we are using a diff sequencer.

File: packages/contractkit/src/test-utils/ganache.setup.ts
Comment (mcortesi) [reply packages/contractkit/src/test-utils/ganache.setup.ts]: Great!
Quick note about it, `waitForPortOpen()` requrires `nc` to be installed. But i guess it's already there for every environment where we run the tests.

**Review by mcortesi — APPROVED** [review]
Looks Great!

I've answered some of the comments from the previous review

**Comment by H34D** [comment]
> Looks Great!
> 
> I've answered some of the comments from the previous review

ill put the pending changes on my list

**Pull request merged**
